### PR TITLE
CLN: remove index keyword from _wrap_aggregated_output

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -371,78 +369,36 @@ def _cython_agg_general(
...
         -----
         In the vast majority of cases output will only contain one element.
         The exception is operations that expand dimensions, like ohlc.
         """
-        result = self._wrap_series_output(output=output, index=index)
+        assert len(output) == 1
+
+        name = self.obj.name
+        index = self.grouper.result_index
+        values = list(output.values())[0]
```
Comment (rhshadrach) [inline pandas/core/groupby/generic.py:399]: Tiny perf boost by not creating the intermediate list

```
d = {'a': 1}
%timeit list(d.values())[0]
%timeit next(iter(d.values()))
```

gives

```
219 ns ± 0.363 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
146 ns ± 22.8 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

**Review by rhshadrach — CHANGES_REQUESTED** [review]
This looks really great, minor comment

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: good idea, updated

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -466,7 +422,10 @@ def _wrap_transformed_output(
         for consistency with DataFrame methods and _wrap_aggregated_output.
         """
         assert len(output) == 1
-        result = self._wrap_series_output(output=output, index=self.obj.index)
+
+        name = self.obj.name
+        values = list(output.values())[0]
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:427]: Looks like the `next` trick can work here too.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
nice!

**Pull request merged**